run_go1.18.go

```diff
@@ -0,0 +1,41 @@
+// Licensed under the MIT license, see LICENSE file for details.
+
+//go:build go1.18
+// +build go1.18
+
+package quicktest
+
+import "testing"
+
+// fastRun implements c.Run for some known types.
+// It returns the result of calling c.Run and also reports
+// whether it was able to do so.
+func fastRun(c *C, name string, f func(c *C)) (bool, bool) {
+	switch t := c.TB.(type) {
+	case runner[*testing.T]:
+		return fastRun1(c, name, f, t), true
+	case runner[*testing.B]:
+		return fastRun1(c, name, f, t), true
+	case runner[*C]:
+		return fastRun1(c, name, f, t), true
```

Comment:
Should we `return fastRun(t, name, f)` here?

Reply:
It could be possible as a shortcut, but maybe a bit too clever. In fact this shortcut assumes that `t` is if of type `*qt.C` which may not be the case (a non-`qt.C` might have a `Run(string, *qt.C) bool` method).

Another alternative which is less clever, but still shortcuts (avoids to create one `qt.C`) while not corrupting the interface:
```suggestion
		return t.Run(name, f), true
```

Note that this shortcut is already implemented in #165.

Reply:
FWIW I thought about this, and even implemented it, but decided it wasn't a good idea. For one, it only saves one allocation, which is a fairly minor saving in this context. For another, it means that it's not consistent with the other types, which always allocate a new C. Making this optimisation changes observable behaviour and because I'm not convinced it's a worthwhile optimisation anyway, I'd be inclined to leave it out.

Reply:
Good catch.

One observable difference would be the [format function](https://pkg.go.dev/github.com/frankban/quicktest#C.SetFormat) that should be propagated from the parent `C`. We obviously lack a test that would have caught this.

Reply:
Test proposed in #167.